glide-core/src/client/value_conversion.rs

```diff
@@ -411,7 +348,25 @@ fn convert_array_elements(
     Ok(Value::Array(converted_array))
 }
 
-fn convert_array_to_map(
+fn convert_inner_map_by_type(
```

Comment:
What do you mean by "inner_map"? It looks like this function converts map keys and values, and those values may or may not also be maps. Would "convert_map_by_type" or "convert_map_entries" be clearer?